src/EdgeFactory/interfaces/IEdgeFactory.sol

```diff
@@ -0,0 +1,110 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+pragma solidity >=0.8.0;
+
+/// @title IEdgeFactory
+/// @custom:security-contact security@euler.xyz
+/// @author Objective Labs (https://www.objectivelabs.io/)
+/// @author Euler Labs (https://www.eulerlabs.com/)
+/// @notice Factory contract for deploying and configuring Edge markets
+interface IEdgeFactory {
+    /// @notice Parameters for deploying a vault
+    /// @param asset The underlying asset of the vault
+    /// @param irm The address of the interest rate model, ignored if escrow
+    /// @param escrow True if the vault is collateral only, false if it is borrowable
+    struct VaultParams {
+        address asset;
+        address irm;
+        bool escrow;
+    }
+
+    /// @notice Parameters for configuring an oracle adapter in the router
+    /// @param base The base token address
+    /// @param quote The quote token address
+    /// @param adapter The oracle adapter address
+    struct AdapterParams {
+        address base;
+        address quote;
```

Comment:
maybe there's no point taking `quote` as a param? it must be equal to the unit of account anyways, otherwise the router will be misconfigured